g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/source/service/impl/SourceServiceImpl.java

```diff
@@ -84,40 +85,52 @@ public void writeSourceToCachedFile() {
 	/**
 	 * merge the string queue source to component Map source
 	 */
-	private void mergeSource2Map() {
+	private boolean mergeSource2Map() {
 
 		StringSourceDTO strDTO = STRING_SOURCES.poll();
 		strDTO.setLocale(ConstantsKeys.LATEST);
-		String key = getKey(strDTO);
+		String key = strDTO.getKey();
 		String source = strDTO.getSource();
 		String comment = strDTO.getComment();
+		String sourceFormat = strDTO.getSourceFormat();
 		String catcheKey = PathUtil.generateCacheKey(strDTO);
 		ComponentSourceDTO comp = PREPARE_MAP.get(catcheKey);
 		
 		if (StringUtils.isEmpty(comp)) {
-			addNewStringSource(strDTO, catcheKey, key, source, comment);
+			addNewStringSource(strDTO, catcheKey, key, source, comment, sourceFormat);
 		} else {
-			updateStringSource(comp, key, source, comment);
+			updateStringSource(comp, key, source, comment, sourceFormat);
 		}
+        if(!StringUtils.isEmpty(sourceFormat) && (MD.equals(sourceFormat) || HTML.equals(sourceFormat))) {
+        	return true;
+        }else {
+        	return false;
+        }
```

Comment:
move these verification to a common method

Reply:
done